tests/core/switch_session_test.py

```diff
@@ -179,3 +205,48 @@ def test_close_session_with_error(self):
 
         with self.assertRaises(UnknownResource):
             self.session_manager.get_switch_for_session('patate')
+
+    def test_add_session(self):
+        self.session_manager.session_storage = flexmock()
+        self.session_manager.session_storage.should_receive('add').with_args(self.switch_mock.switch_descriptor, 'patate')
+
+        self.session_manager.add_session(self.switch_mock, 'patate')
+
+        assert_that(self.session_manager.sessions['patate'], is_(self.switch_mock))
+
+    def test_add_session_raises_if_already_exists(self):
+        self.session_manager.session_storage = flexmock()
+        self.session_manager.sessions['patate'] = self.switch_mock
+        self.session_manager.session_storage.should_receive('add').never()
+
+        with self.assertRaises(SessionAlreadyExists):
+            self.session_manager.add_session(self.switch_mock, 'patate')
```

Comment:
if you agree with my opninon that this is a private method, then you should be testing using open_session